### PR TITLE
Add AWS Lambda Support

### PR DESCRIPTION
**Pull request by Manouchehri**
Technically this also supports Lambda@Edge, but in practice that isn't a great idea, since you're capped to 30 seconds with Lambda@Edge. https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/cloudfront-limits.html#limits-lambda-at-edge

Using Lambda seems to work fine in my limited testing tonight. https://q56f63qclhddf4zdmg7wksnrmq0ytyyt.lambda-url.us-east-1.on.aws/ (This URL may be offline by the time you read this in the future.)

To deploy, run:

```sh
sam build && sam deploy --guide
```

### CODE REVIEW

**Comment by Manouchehri** [comment]
Note: we aren't using arm64 (despite being cheaper/faster), because psutil isn't compiled for it.

```
dave@mbp litellm % sam build
Building codeuri: /Users/dave/Work/litellm/litellm runtime: python3.11 metadata: {} architecture: arm64 functions: Function
 Running PythonPipBuilder:ResolveDependencies

Build Failed
Error: PythonPipBuilder:ResolveDependencies - {psutil==5.9.6(wheel)}
```

x86_64 works fine as expected.

**Comment by ghost** [comment]
@Manouchehri this is amazing - thank you for your PR! 

Is there anything we can add to docs that would be helpful for someone trying to do what you did?

**Comment by Manouchehri** [comment]
Just `sam build && sam deploy --guide` should be enough, along with a link to https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/install-sam-cli.html

tl;dr:

```sh
# Install SAM - https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/install-sam-cli.html
git clone https://github.com/BerriAI/litellm.git
cd litellm/
sam build
sam deploy --guide
```

**Comment by sangshuduo** [comment]
seems template.yaml is removed. does it mean lambda will not be supported officially?

**Comment by Manouchehri** [comment]
@sangshuduo I ended up not using Lambda in production for LiteLLM, so I'm not sure if anybody is testing it anymore.

**Comment by sangshuduo** [comment]
> @sangshuduo I ended up not using Lambda in production for LiteLLM, so I'm not sure if anybody is testing it anymore.

I'm interesting on run litellm as a lambda function. I use FastAPI to take request then use litellm to forward requests to chatGPT. However, I always get annoying TooManyRedirects error from lambda function url. I wonder if there is an official doc and sample code to demo how to integrate them together.
